spec/chunky_png/datastream_spec.rb

```diff
@@ -58,4 +59,60 @@
       expect{physical_chunk.dpiy}.to raise_error(ChunkyPNG::UnitsUnknown)
     end
   end
+
+  describe '#iTXt_chunk' do
+    it 'should read iTXt chunks correctly' do
+      filename = resource_file('itxt_chunk.png')
+      ds = ChunkyPNG::Datastream.from_file(filename)
+      int_text_chunks = ds.chunks.select { |chunk|
+        chunk.is_a?(ChunkyPNG::Chunk::InternationalText)
+      }
+      expect(int_text_chunks.length).to eq(2)
+
+
+      coach_uk = int_text_chunks.find { |chunk|
+        chunk.language_tag == 'en-gb'
+      }
+      coach_us = int_text_chunks.find { |chunk|
+        chunk.language_tag == 'en-us'
+      }
+      expect(coach_uk).to_not be_nil
+      expect(coach_us).to_not be_nil
+
+      expect(coach_uk.keyword).to eq('coach')
+      expect(coach_uk.text).to eq('bus with of higher standard of comfort, usually chartered or used for longer journeys')
+      expect(coach_uk.translated_keyword).to eq('bus')
+      expect(coach_uk.compressed).to eq(ChunkyPNG::UNCOMPRESSED_CONTENT)
+      expect(coach_uk.compression).to eq(ChunkyPNG::COMPRESSION_DEFAULT)
+
+      expect(coach_us.keyword).to eq('coach')
+      expect(coach_us.text).to eq('US extracurricular sports teacher at a school (UK: PE teacher) lowest class on a passenger aircraft (UK: economy)')
+      expect(coach_us.translated_keyword).to eq('trainer')
+      expect(coach_us.compressed).to eq(ChunkyPNG::COMPRESSED_CONTENT)
+      expect(coach_us.compression).to eq(ChunkyPNG::COMPRESSION_DEFAULT)
+    end
+
+    it 'should write iTXt chunks correctly' do
+      expected_hex = %w(0000 001d 6954 5874 436f 6d6d 656e 7400 0000 0000 4372 6561 7465 6420 7769 7468 2047 494d 5064 2e65 07).join('')
+      stream = StringIO.new
+      itext = ChunkyPNG::Chunk::InternationalText.new('Comment', 'Created with GIMP')
+      itext.write(stream)
+      generated_hex = stream.string.unpack('H*').join('')
+
+      expect(generated_hex).to eq(expected_hex)
+    end
+
+    it 'should handle UTF-8 in iTXt chunks correctly' do
+      stream = StringIO.new
+      text = '☼☕'.force_encoding('utf-8')
+      translated_keyword = '⚡'.force_encoding('utf-8')
+      itext = ChunkyPNG::Chunk::InternationalText.new('Comment', text, '', translated_keyword)
+      itext.write(stream)
+
+      parsed = ChunkyPNG::Chunk.read(StringIO.new(stream.string))
+      expect(parsed.keyword).to eq('Comment')
+      expect(parsed.text).to eq(text)
+      expect(parsed.translated_keyword).to eq(translated_keyword)
```

Comment:
I'd add `expect(parsed.text.encoding).to eq(Encoding::UTF_8)` here. Not only does it make sure it's working and the encoding doesn't get lost, but also documents how it's supposed to work.